### ISSUE TITLE
Issue with --save_resume option 

### ISSUE THREAD

**Issue by ramab1988** [VowpalWabbit/vowpal_wabbit#1672]
Hi Guys!

I am using VW to build a text classification model & I would like to use it in an online fashion (continue training on new data as and when they come in). In the first step I am training on a sample training dataset (train.vw) & saving the model (model.vw). Before training again, I am testing it on a test set (& calculate AUC). In the following commands, training works perfectly fine, but while testing, the saved model is throwing an error. Could you please help me with this?

### Training command
``vw train.vw -c -k --passes 20 --save_resume --learning_rate=0.1 --loss_function logistic --link=logistic -b 29 -q to --l1 1e-6 --l2 1e-6 -f model.vw``

### Testing command
``vw test.vw -t -i model.vw -p preds.txt``

### Error
``Disagreeing option values for '--l1_state': '0.000804683024372347' vs '0.00165470062766786'
vw (parser_helper.cc:172): Disagreeing option values for '--l1_state': '0.000804683024372347' vs '0.00165470062766786'``

### I observed following things

1. If I don't give the option --save_resume, the model trains perfectly fine & testing also runs without any issues. But then I will not be able to do online learning. 
2. If I don't give l1 & l2 regularization, then the above mentioned error will not come up, but the AUC is pretty bad.

### Version & other details
I am running VW 8.6.1 on Mac installed using brew

**Comment by JohnLangford** [comment]
Can you make this reproducible?

**Comment by ramab1988** [comment]
Thanks for your reply John. Attaching train.txt & test.txt files for which i am getting above error. 
[train.txt](https://github.com/VowpalWabbit/vowpal_wabbit/files/2558705/train.txt)
[test.txt](https://github.com/VowpalWabbit/vowpal_wabbit/files/2558706/test.txt)


### Training
`vw train.txt -c -k --passes 20 --save_resume --learning_rate=0.1 --loss_function logistic --link=logistic -b 29 -q to --l1 1e-6 --l2 1e-6 -f model.vw`

### Testing
`vw test.txt -t -i model.vw -p preds.txt`

Version & other details
I am running VW 8.6.1 on Mac installed using brew. 

Please let me know if you need any other details. 

On a different note, I have started using python interface & it is not throwing any error while saving / loading the model. But I have an important question here: Is it thread safe? What happens if multiple threads start running learn/predict on the same VW object at the same time?

Thanks a lot in advance!

**Comment by lokitoth** [comment]
Is this a duplicate of #1573?

**Comment by JohnLangford** [comment]
Yes.   It looks like a bad interaction with the extra state required to get numerical precision to work right.

**Comment by ramab1988** [comment]
I guess so. Thanks for pointing it out. Is there any fix for this?

**Comment by jackgerrits** [comment]
Assigned to @pavithraks to check if it still repros with the options patch

**Comment by pavithraks** [comment]
Somehow, the changes in https://github.com/VowpalWabbit/vowpal_wabbit/pull/1706 did not fix this. I still see the error when I run the example data that @ramab1988 provided. We marked this one as a duplicate of 1573.  I'm looking into this. Let's not mark this as closed for now.

**Comment by pavithraks** [comment]
okay, false alarm. I was running the installed vw on linux instead of the one with the fix (facepalm). The save_resume issue has been fixed. Although, I don't know the answer to @ramab1988 's other question about threads. (copied below)

_On a different note, I have started using python interface & it is not throwing any error while saving / loading the model. But I have an important question here: Is it thread safe? What happens if multiple threads start running learn/predict on the same VW object at the same time?_

maybe @jackgerrits or @JohnLangford  can respond to that?
